extensions/amp-ima-video/0.1/test/validator-amp-ima-video.html

```diff
@@ -34,5 +34,12 @@
       data-tag="https://pubads.g.doubleclick.net/gampad/ads?sz=640x480&iu=/124319096/external/ad_rule_samples&ciu_szs=300x250&ad_rule=1&impl=s&gdfp_req=1&env=vp&output=vmap&unviewed_position_start=1&cust_params=deployment%3Ddevsite%26sample_ar%3Dpremidpost&cmsid=496&vid=short_onecue&correlator="
       data-poster="path/to/poster.png">
   </amp-ima-video>
+
+  <amp-ima-video width=640 height=360 layout="responsive" autoplay data-tag="https://pubads.g.doubleclick.net/gampad/ads?sz=640x480&iu=/124319096/external/ad_rule_samples&ciu_szs=300x250&ad_rule=1&impl=s&gdfp_req=1&env=vp&output=vmap&unviewed_position_start=1&cust_params=deployment%3Ddevsite%26sample_ar%3Dpremidpost&cmsid=496&vid=short_onecue&correlator="
+    data-poster="path/to/poster.png">
+    <source src="https://s0.2mdn.net/4253510/google_ddm_animation_480P.mpeg" type="application/x-mpegURL">
+    <source src="https://s0.2mdn.net/4253510/google_ddm_animation_480P.mp4" type="video/mp4"> >
+    <track kind="captions" srclang="en" src="https://s0.2mdn.net/4253510/google_ddm_animation_480P.mp4.track">
```

Comment:
Just to confirm: `preconnectCallback` doesn't happen during prerender, right? Because https://github.com/ampproject/amphtml/blob/master/extensions/amp-ima-video/0.1/amp-ima-video.js builds preconnects out of these `src` URLs.

Reply:
`preconnectCallback` does get called during prerender but it is just collecting data, `preconnect.url()` is async and waits for visibility of the document before running (https://github.com/ampproject/amphtml/blob/4a9ec7291763d35789b6e31a40b6d3b2c13ccb80/src/preconnect.js#L130)